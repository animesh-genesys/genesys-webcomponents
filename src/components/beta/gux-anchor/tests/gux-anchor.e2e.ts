import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-anchor-beta', () => {
  describe('#render', () => {
    [
      '<gux-anchor-beta><a href="#">Default Link</a></gux-anchor-beta>',
      '<gux-anchor-beta><a href="#" target="_blank">Open in new tab</a></gux-anchor-beta>',
      '<gux-anchor-beta><a href="https://spark.genesys.com/" target="_blank">External Link</a></gux-anchor-beta>',
      '<gux-anchor-beta table="true"><a href="#">Tabel Cell Link</a></gux-anchor-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-anchor-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
