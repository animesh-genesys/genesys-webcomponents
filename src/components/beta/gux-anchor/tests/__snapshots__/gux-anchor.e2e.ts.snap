// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-anchor-beta #render should render component as expected (1) 1`] = `"<gux-anchor-beta table=\\"false\\" hydrated=\\"\\"><a href=\\"#\\">Default Link</a></gux-anchor-beta>"`;

exports[`gux-anchor-beta #render should render component as expected (2) 1`] = `"<gux-anchor-beta table=\\"false\\" hydrated=\\"\\"><a href=\\"#\\" target=\\"_blank\\">Open in new tab</a></gux-anchor-beta>"`;

exports[`gux-anchor-beta #render should render component as expected (3) 1`] = `"<gux-anchor-beta table=\\"false\\" hydrated=\\"\\"><a href=\\"https://spark.genesys.com/\\" target=\\"_blank\\">External Link</a></gux-anchor-beta>"`;

exports[`gux-anchor-beta #render should render component as expected (4) 1`] = `"<gux-anchor-beta table=\\"true\\" hydrated=\\"\\"><a href=\\"#\\">Tabel Cell Link</a></gux-anchor-beta>"`;
